include/mockturtle/networks/aqfp.hpp

```diff
@@ -0,0 +1,1278 @@
+    ++_storage->data.num_pis;
+    return { index, 0 };
+  }
+
+  uint32_t create_po( signal const& f, std::string const& name = std::string() )
+  {
+    (void)name;
+
+    /* increase ref-count to children */
+    _storage->nodes[f.index].data[0].h1++;
+    auto const po_index = static_cast<uint32_t>( _storage->outputs.size() );
+    _storage->outputs.emplace_back( f.index, f.complement );
+    ++_storage->data.num_pos;
+    return po_index;
+  }
+
+  signal create_ro( std::string const& name = std::string() )
+  {
+    (void)name;
+
+    auto const index = _storage->nodes.size();
+    auto& node = _storage->nodes.emplace_back();
+    node.children.resize( 3u );
+    node.children[0].data = node.children[1].data = node.children[2].data = _storage->inputs.size();
+    _storage->inputs.emplace_back( index );
+    return { index, 0 };
+  }
+
+  uint32_t create_ri( signal const& f, int8_t reset = 0, std::string const& name = std::string() )
+  {
+    (void)name;
+
+    /* increase ref-count to children */
+    _storage->nodes[f.index].data[0].h1++;
+    auto const ri_index = static_cast<uint32_t>( _storage->outputs.size() );
+    _storage->outputs.emplace_back( f.index, f.complement );
+    _storage->data.latches.emplace_back( reset );
+    return ri_index;
+  }
+
+  int8_t latch_reset( uint32_t index ) const
+  {
+    assert( index < _storage->data.latches.size() );
+    return _storage->data.latches[index];
+  }
+
+  bool is_combinational() const
+  {
+    return ( static_cast<uint32_t>( _storage->inputs.size() ) == _storage->data.num_pis &&
+             static_cast<uint32_t>( _storage->outputs.size() ) == _storage->data.num_pos );
+  }
+
+  bool is_constant( node const& n ) const
+  {
+    return n == 0;
+  }
+
+  bool is_ci( node const& n ) const
+  {
+    return _storage->nodes[n].children[0].data == _storage->nodes[n].children[1].data && _storage->nodes[n].children[0].data == _storage->nodes[n].children[2].data;
+  }
+
+  bool is_pi( node const& n ) const
+  {
+    return _storage->nodes[n].children[0].data == ~static_cast<uint64_t>( 0 ) && _storage->nodes[n].children[1].data == ~static_cast<uint64_t>( 0 ) && _storage->nodes[n].children[2].data == ~static_cast<uint64_t>( 0 );
+  }
+
+  bool is_ro( node const& n ) const
+  {
+    return _storage->nodes[n].children[0].data == _storage->nodes[n].children[1].data && _storage->nodes[n].children[0].data == _storage->nodes[n].children[2].data && _storage->nodes[n].children[0].data >= static_cast<uint64_t>( _storage->data.num_pis );
+  }
+
+  bool constant_value( node const& n ) const
+  {
+    (void)n;
+    return false;
+  }
+#pragma endregion
+
+#pragma region Create unary functions
+  signal create_buf( signal const& a )
+  {
+    return a;
+  }
+
+  signal create_not( signal const& a )
+  {
+    return !a;
+  }
+#pragma endregion
+
+#pragma region Create binary / ternary functions
+  signal create_maj( signal a, signal b, signal c )
+  {
+    return create_maj( { a, b, c } );
```

Comment:
I might be mistaken here, but doesn't this line cause the rest of the function code to be unreachable?

Reply:
Thanks. It is fixed now.